src/main/java/org/dataloader/MappedBatchObserver.java

```diff
@@ -0,0 +1,34 @@
+package org.dataloader;
+
+/**
+ * A interface intended as a delegate for other Observer-like classes used in other libraries, to be invoked by the calling
+ * {@link MappedObserverBatchLoader}.
+ * <p>
+ * Some examples include:
+ * <ul>
+ *     <li>Project Reactor's <a href="https://www.reactive-streams.org/reactive-streams-1.0.4-javadoc/org/reactivestreams/Subscriber.html">{@code Subscriber}</a>
+ *     <li>gRPC's <a href="https://grpc.github.io/grpc-java/javadoc/io/grpc/stub/StreamObserver.html">{@code StreamObserver}</a>
+ *     <li>RX Java's <a href="https://reactivex.io/RxJava/3.x/javadoc/io/reactivex/rxjava3/core/Flowable.html">{@code Flowable}</a>
+ * </ul>
+ * @param <K> the key type of the calling {@link MappedObserverBatchLoader}.
+ * @param <V> the value type of the calling {@link MappedObserverBatchLoader}.
+ */
+public interface MappedBatchObserver<K, V> {
+
+    /**
+     * To be called by the {@link MappedObserverBatchLoader} to process a new key/value pair.
+     */
+    void onNext(K key, V value);
```

Comment:
This is the same class as `BatchObserver<K, V>` except onNext has a key and value

How about making it

```
public interface MappedBatchObserver<K, V> extends BatchObserver<K, V>  {

    void onNext(java.util.Map.Entry<K,V> nextItem);

```

?? (if that compiles)

Reply:
This maybe reduce the amount of code you need to handle the two AND also maps on reactive streams `Publisher` better which is only ever 1 value

Reply:
Yep, I've elected to deprecate both in favour of:

 - `Publisher<V>`
 - `Publisher<Map.Entry<K, V>>`